polkadot-test-service/tests/build-blocks.rs

```diff
@@ -0,0 +1,68 @@
+use futures::{future, pin_mut, select, FutureExt as _, StreamExt};
+use polkadot_test_service::*;
+use service::TaskType;
+use std::pin::Pin;
+use std::sync::Arc;
+use sp_keyring::Sr25519Keyring;
+use sc_client_api::client::BlockchainEvents;
+use std::collections::HashSet;
+
+fn task_executor(
+) -> Arc<dyn Fn(Pin<Box<dyn futures::Future<Output = ()> + Send>>, TaskType) + Send + Sync> {
+	Arc::new(
+		move |fut: Pin<Box<dyn futures::Future<Output = ()> + Send>>, _| {
+			async_std::task::spawn(fut.unit_error());
+		},
+	)
+}
+
+#[async_std::test]
+async fn ensure_test_service_build_blocks() {
+	sc_cli::init_logger("");
+	let (service_alice, client_alice, _handles_alice, _base_path_alice) = run_test_node(task_executor(), 27015, Sr25519Keyring::Alice, || {
```

Comment:
I assume this is the port? This should not be required and we should select this randomly.

Reply:
And we should return one struct that contains all this data.

Reply:
done and done